### ISSUE TITLE
Falling through floor with low FPS

### ISSUE THREAD

**Issue by douglas-moyes** [tsunamayo/Starship-EVO#5032]
### Short Description of the issue:

Low FPS causes the avatar to fall through the floor and out into space.

### Steps to reproduce the issue:
Run on Linux with Proton at 4k, turn off jetpack in space station and fall... fall down through the floor.

Adjust graphics settings to a pathetically low value so FPS is tolerable, turn off jetpack, and you'll land on the floor.
...

### Environment
- Starship EVO version: 22w38a
- Graphics Card:  nVidia 1660 Ti

### Additional information:
This really needs to be optimized for Proton, especially if you want it to work on Steam Deck.
For now though, Linux is a good place to test to see what happens to the game when it lags like crazy.

**Comment by douglas-moyes** [comment]
Looks like collision detection handling is tied to the same code that handles the screen/scene presentation. Collision and object positions are best handled in their own thread independent of screen updates. It just needs to be smart about checking if an object passed through another and make the appropriate actions (move the object, or trigger a collision/damage effect). You'll likely see a screen rendering performance increase since the thread handing screen updates is only doing screen updates and nothing else.

**Comment by tsunamayo** [comment]
@douglas-moyes Yes it is a known issue, it is why there is a minimum required config and I dont support Linux. Thanks
